apiserver/httpcontext.go

```diff
@@ -65,6 +67,16 @@ func (ctxt *httpContext) controllerObjectStoreForRequest(r *http.Request) (objec
 	return ctxt.srv.shared.objectStoreGetter.GetObjectStore(r.Context(), st.ControllerModelUUID())
 }
 
+// serviceFactoryForRequest returns a service factory for a
+// given request.
+func (ctxt *httpContext) serviceFactoryForRequest(r *http.Request) (servicefactory.ServiceFactory, error) {
```

Comment:
I can only see one usage of this, maybe we can just inline it for now?

Reply:
We didn't inline it to make it easily available to the apiserver as more work is done to starting using services there rather than state.